src/coreclr/jit/gentree.cpp

```diff
@@ -1603,6 +1603,38 @@ CallArg* CallArgs::GetArgByIndex(unsigned index)
     return cur;
 }
 
+//---------------------------------------------------------------
+// GetUserArgByIndex: Get an argument with the specified index.
+//   Unlike GetArgByIndex, this function ignores non-user args
+//   like r2r cells.
+//
+// Parameters:
+//   index - The index of the argument to find.
+//
+// Returns:
+//   A pointer to the argument.
+//
+// Remarks:
+//   This function assumes enough arguments exist. The current implementation
+//   doesn't ignore return buffers and type arguments.
+//
+CallArg* CallArgs::GetUserArgByIndex(unsigned index)
+{
+    CallArg* cur = m_head;
+    assert((cur != nullptr) && "Not enough user arguments in GetUserArgByIndex");
+    for (unsigned i = 0; i < index || cur->IsArgAddedLate();)
+    {
+        if (!cur->IsArgAddedLate())
```

Comment:
Should this use `cur->GetWellKnownArg() != WellKnownArg::None` instead? (And then the Remarks "The current implementation doesn't..." can be removed?)

Reply:
It would ignore the 'this' arg and also ShiftLow/ShiftHigh for x86 shift helpers are arguably "user args" (in that they manifest in the IL). Maybe we should add an `IsUserArg()` or `IsILArg()` that has these special cases and use it here.

Reply:
Added `IsUserArg()`